### DISCUSSION TITLE
Passing training step in the forward() of allennlp model

### DISCUSSION THREAD

**Discussion by awasthiabhijeet** [allenai/allennlp#5349] [Q&A]
Hi,

I wish to use the training step as a parameter to change dropout probability inside the forward() function in the implementation of my allennlp model. 
What would be the best way to do it? 
I want to make use of `allenlp.train`, and wish to avoid writing a custom training loop myself.

Thanks,

**Comment by epwalsh** [comment]
If I understand correctly, you want to change the dropout probability of your model throughout training depending on how many steps you've trained for so far?

You could accomplish this with a custom `TrainerCallback` using the `on_batch` method: https://docs.allennlp.org/main/api/training/callbacks/callback/#on_batch.

**Reply by awasthiabhijeet** [reply]
Thanks for the prompt reply :)
Yes, I want to make dropout probability a function of the training steps.
